cookbooks/vectortile/recipes/default.rb

```diff
@@ -0,0 +1,290 @@
+#
+# Cookbook:: vectortile
+# Recipe:: default
+#
+# Copyright:: 2024, OpenStreetMap Foundation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+include_recipe "accounts"
+include_recipe "git"
+include_recipe "nginx"
+include_recipe "postgresql"
+include_recipe "prometheus"
+include_recipe "python"
+include_recipe "tools"
+
+directory "/srv/vector.openstreetmap.org" do
+  user "tileupdate"
+  group "tileupdate"
+  mode "755"
+end
+
+nginx_site "default" do
+  action [:delete]
+end
+
+nginx_site "vector.openstreetmap.org" do
+  template "nginx.erb"
+end
+
+ssl_certificate node[:fqdn] do
+  domains [node[:fqdn], "vector.openstreetmap.org"]
+  notifies :reload, "service[nginx]"
+end
+
+remote_directory "/srv/vector.openstreetmap.org/html" do
+  source "html"
+  owner "www-data"
+  group "www-data"
+  mode "755"
+  files_owner "www-data"
+  files_group "www-data"
+  files_mode "644"
+end
+
+template "/srv/vector.openstreetmap.org/html/index.html" do
+  source "index.html.erb"
+  owner "www-data"
+  group "www-data"
+  mode "644"
+end
+
+postgresql_version = node[:vectortile][:database][:cluster].split("/").first
+postgis_version = node[:vectortile][:database][:postgis]
+
+package "postgresql-#{postgresql_version}-postgis-#{postgis_version}"
+
+# Get a recent osm2pgsql version with backports.
+apt_preference "osm2pgsql" do
+  pin "release o=Debian Backports"
+  pin_priority "600"
+end
```

Comment:
This shouldn't be necessary now - the pinning is now handled in the `apt::default` recipe and includes osm2pgsql for Debian 12 hosts.